camel/src/main/java/com/github/theprez/manzan/configuration/DataConfig.java

```diff
@@ -104,7 +91,62 @@ public synchronized Map<String, ManzanRoute> getRoutes() throws IOException, AS4
                     throw new RuntimeException("Unknown destination type: " + type);
             }
         }
+
+        // We will create a formatMap to store the format for each watch session, as well
+        // as a destMap to store the destinations for each watch session
+        final Map<String, String> formatMap = new HashMap<>();
+        final Map<String, String> destMap = new HashMap<>();
+
+        for (int i = 0; i < watchEvents.size(); i++) {
+            final String section = watchEvents.get(i);
+            final String name = section;
+            int userNumToProcess = getOptionalInt(name, "numToProcess");
+            final int numToProcess = userNumToProcess != -1 ? userNumToProcess : DEFAULT_NUM_TO_PROCESS;
+            final String format = getOptionalString(name, "format");
+            String strwch = getRequiredString(name, "strwch");
+            String id = getRequiredString(name, "id");
+
+            ManzanEventType eventType;
+            if(strwch.contains("WCHMSGQ")) {
+                eventType = ManzanEventType.WATCH_MSG;
+            } else if(strwch.contains("WCHLICLOG")) {
+                eventType = ManzanEventType.WATCH_VLOG;
+            } else if(strwch.contains("WCHPAL")) {
+                eventType = ManzanEventType.WATCH_PAL;
+            } else {
+                throw new RuntimeException("Watch for message, LIC log entry, or PAL entry not specified");
+            }
+
+            int userInterval = getOptionalInt(name, "interval");
+            final int interval = userInterval != -1 ? userInterval : DEFAULT_INTERVAL;
+            final List<String> destinations = new LinkedList<String>();
+            for (String d : getRequiredString(name, "destinations").split("\\s*,\\s*")) {
+                d = d.trim();
+                if (!m_destinations.contains(d)) {
+                    throw new RuntimeException(
+                            "No destination configured named '" + d + "' for data source '" + name + "'");
+                }
+                if (StringUtils.isNonEmpty(d)) {
+                    destinations.add(d);
+                }
+            }
+
+            // Build the maps
+            String destString = createRecipientList(destinations);
+            formatMap.put(id.toUpperCase(), format);
+            destMap.put(id.toUpperCase(), destString);
+
+            String sqlRouteName = name + "sql";
+            ret.put(sqlRouteName, new WatchMsgEventSql(sqlRouteName, id, format, destinations, schema, eventType, interval, numToProcess));
+            WatchStarter ws = new WatchStarter(id, strwch);
+            ws.strwch();
+
+            if (i == watchEvents.size() - 1){
```

Comment:
Can we just move this outside the look and do this if `watchEvents.size() > 0` to avoid having to check this on each iteration?

Reply:
Moved after for loop now